File: src/main/java/smart4aviation/repository/JsonRepository.java
package smart4aviation.repository;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.AllArgsConstructor;
import org.apache.commons.io.IOUtils;
import org.springframework.core.io.ClassPathResource;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.util.List;

@AllArgsConstructor
public abstract class JsonRepository<T> {

    private String mockPath;
    private TypeReference<List<T>> typeReference;
    private ObjectMapper objectMapper;

    public List<T> findAll() throws IOException {
        return findAllByPathAndType(mockPath, typeReference);
    }

    private List<T> findAllByPathAndType(String path, TypeReference<List<T>> typeReference) throws IOException {
        ClassPathResource classPathResource = new ClassPathResource(path);
        String staticDataString = IOUtils.toString(classPathResource.getInputStream(), StandardCharsets.UTF_8);
        return objectMapper.readValue(staticDataString, typeReference);
    }

}

File: src/main/kotlin/com/example/smart4aviation/repository/JsonRepository.kt
package com.example.smart4aviation.repository

import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.ObjectMapper
import org.apache.commons.io.IOUtils
import org.springframework.core.io.ClassPathResource
import java.nio.charset.StandardCharsets


open class JsonRepository<T>(
    private val mockPath: String,
    private val typeReference: TypeReference<List<T>>,
    private val objectMapper: ObjectMapper
) {

    fun findAll(): List<T> {
        return findAllByPathAndType(mockPath, typeReference)
    }

    private fun findAllByPathAndType(path: String, typeReference: TypeReference<List<T>>): List<T> {
        val classPathResource = ClassPathResource(path)
        val staticDataString: String = IOUtils.toString(classPathResource.inputStream, StandardCharsets.UTF_8)
        return objectMapper.readValue(staticDataString, typeReference)
    }

}